fcrepo-kernel-impl/src/test/java/org/fcrepo/kernel/impl/TransactionManagerImplTest.java

```diff
@@ -88,8 +97,108 @@ public void testGetTransactionWithInvalidID() {
     }
 
     @Test(expected = TransactionRuntimeException.class)
-    public void testGetExpiredTransaction() {
+    public void testGetExpiredTransaction() throws Exception {
         testTx.expire();
-        testTxManager.get(testTx.getId());
+        try {
+            testTxManager.get(testTx.getId());
+        } finally {
+            // Make sure rollback is triggered
+            verify(psSession).rollback();
+        }
+    }
+
+    @Test
+    public void testCleanupClosedTransactions() {
+        System.setProperty(TransactionImpl.TIMEOUT_SYSTEM_PROPERTY, "10000");
+
+        final var commitTx = testTxManager.create();
+        commitTx.commit();
+        final var continuingTx = testTxManager.create();
+        final var rollbackTx = testTxManager.create();
+        rollbackTx.rollback();
+
+        // verify that transactions retrievable before cleanup
+        try {
+            testTxManager.get(commitTx.getId());
+            fail("Transaction must be committed");
+        } catch(final TransactionClosedException e) {
+            //expected
+        }
+        try {
+            testTxManager.get(rollbackTx.getId());
+            fail("Transaction must be rolled back");
+        } catch(final TransactionClosedException e) {
+            //expected
+        }
+
+        assertNotNull("Continuing transaction must be present",
+                testTxManager.get(continuingTx.getId()));
+
+        testTxManager.cleanupClosedTransactions();
+
+        // Verify that the closed transactions are stick around since they haven't expired yet
```

Comment:
I see now that this we are hanging on to transaction regardless of state until the clean up period. Just wondering if there is any value to having those transactions that have been actively committed or rolled back?

Reply:
If there are multiple clients using the same transaction it might be helpful to be able to check the status, which we aren't disallowing at the moment. Figured it'd be good to have a consistent window for being able to look up what happened to a transaction